sdk/nodejs/__tests__/cloudClient.spec.ts

```diff
@@ -0,0 +1,127 @@
+            const res = await client.variable(user, 'test-key-not-in-config', false)
+            expect(res.value).toBe(false)
+            expect(res.isDefaulted).toBe(true)
+        })
+
+        it('to throw an error if key is not defined', async () => {
+            try {
+                await client.variable(user, undefined as unknown as string, false)
+            } catch (ex) {
+                expect(ex.message).toBe(
+                    'Missing parameter: key'
+                )
+            }
+        })
+
+        it('to throw an error if defaultValue is not defined', async () => {
+            try {
+                await client.variable(user, 'test-key', undefined as unknown as string)
+            } catch (ex) {
+                expect(ex.message).toBe(
+                    'Missing parameter: defaultValue'
+                )
+            }
+        })
+    })
+
+    describe('allVariables', () => {
+        it('to return all variables', async () => {
+            const res = await client.allVariables(user)
+            expect(res).toEqual({
+                'test-key': {
+                    key: 'test-key',
+                    value: true,
+                    type: 'Boolean',
+                    defaultValue: false
+                }
+            })
+        })
+
+        it('to return no variables when allVariables call succeeds but user has no variables', async () => {
+            const res = await client.allVariables(emptyUser)
+            expect(res).toEqual({})
+        })
+
+        it('to return no variables when allVariables call fails', async () => {
+            const res = await client.allVariables(badUser)
+            expect(res).toEqual({})
+        })
+    })
+
+    describe('allFeatures', () => {
+        it('to return all features', async () => {
+            const res = await client.allFeatures(user)
+            expect(res).toEqual({
+                'test-feature': {
+                    _id: 'test-id',
+                    _variation: 'variation-id',
+                    variationKey: 'variationKey',
+                    variationName: 'Variation Name',
+                    key: 'test-feature',
+                    type: 'release',
+                }
+            })
+        })
+
+        it('to return no features when allFeatures call succeeds but user has no features', async () => {
+            const res = await client.allFeatures(emptyUser)
+            expect(res).toEqual({})
+        })
+
+        it('to return no features when allFeatures call fails', async () => {
+            const res = await client.allFeatures(badUser)
+            expect(res).toEqual({})
+        })
+    })
+
+    describe('track', () => {
+        const event = {
+            type: 'test-event',
+        }
+        it('an event successfully', async () => {
+            try {
+                client.track(user, event)
+            } catch (ex) {
+                expect(ex).not.toHaveBeenCalled()
```

Comment:
i dont think this tests that no exception is thrown

Reply:
need to investigate further into how to do this then...